v2/grpcworkload/option.go

```diff
@@ -0,0 +1,13 @@
+package grpcworkload
+
+import "github.com/spiffe/go-spiffe/v2/workloadapi"
+
+// Options are used to influence optional behavior. An Option is also a
+// DialOption.
+type Option interface{}
```

Comment:
Probably makes sense to include all option types in options.go? Like DialOptions and ServerOptions?

On naming, I commented earlier about convention around Options. Does it make sense for DialOptions to be ClientOptions instead? I know that DialOptions comes from gRPC, but consistency in our own exposed interfaces would also be good

Reply:
I'd be fine dumping all options together in the same source file.

Regarding Client v.s. Dial options, I think it consistent to have `Option` be used for generally applicable options or when there is only one kind of "operation" or "target" of the options. Otherwise, if there are more targets, I thought it consistent to align the options based on the target of the option (i.e. dialing gets `DialOption`, creating a server gets `ServerOption`). It feels weird to me to pass `ClientOption` into `Dial()` if the only client-like operation we have is `Dial()`-ing. If we were to grow and get more client-like operations, then maybe `ClientOption` wouldn't be the right choice unless not all of those options made sense for dialing.

If that doesn't smell good though, I'm very open to coming up with a better rule-of-thumb.